src/System.ComponentModel.TypeConverter/src/System/Timers/Timer.cs

```diff
@@ -0,0 +1,352 @@
+                    IDesignerHost host = (IDesignerHost)GetService(typeof(IDesignerHost));
+                    if (host != null)
+                    {
+                        object baseComponent = host.RootComponent;
+                        if (baseComponent != null && baseComponent is ISynchronizeInvoke)
+                            _synchronizingObject = (ISynchronizeInvoke)baseComponent;
+                    }
+                }
+
+                return _synchronizingObject;
+            }
+
+            set
+            {
+                _synchronizingObject = value;
+            }
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Notifies
+        ///       the object that initialization is beginning and tells it to stand by.
+        ///    </para>
+        /// </summary>
+        public void BeginInit()
+        {
+            Close();
+            _initializing = true;
+        }
+
+        /// <summary>
+        ///    <para>Disposes of the resources (other than memory) used by
+        ///       the <see cref='System.Timers.Timer'/>.</para>
+        /// </summary>
+        public void Close()
+        {
+            _initializing = false;
+            _delayedEnable = false;
+            _enabled = false;
+
+            if (_timer != null)
+            {
+                _timer.Dispose();
+                _timer = null;
+            }
+        }
+
+        protected override void Dispose(bool disposing)
+        {
+            Close();
+            _disposed = true;
+            base.Dispose(disposing);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Notifies the object that initialization is complete.
+        ///    </para>
+        /// </summary>
+        public void EndInit()
+        {
+            _initializing = false;
+            Enabled = _delayedEnable;
+        }
+
+        /// <summary>
+        /// <para>Starts the timing by setting <see cref='System.Timers.Timer.Enabled'/> to <see langword='true'/>.</para>
+        /// </summary>
+        public void Start()
+        {
+            Enabled = true;
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Stops the timing by setting <see cref='System.Timers.Timer.Enabled'/> to <see langword='false'/>.
+        ///    </para>
+        /// </summary>
+        public void Stop()
+        {
+            Enabled = false;
+        }
+
+        private void MyTimerCallback(object state)
+        {
+            // System.Threading.Timer will not cancel the work item queued before the timer is stopped.
+            // We don't want to handle the callback after a timer is stopped.
+            if (state != _cookie)
+            {
+                return;
+            }
+
+            if (!_autoReset)
+            {
+                _enabled = false;
+            }
+
+            ElapsedEventArgs elapsedEventArgs = new ElapsedEventArgs(DateTime.UtcNow.ToFileTime());
+            try
```

Comment:
This implementation is changed from Desktop. Desktop implementation [here](http://referencesource.microsoft.com/#System/services/timers/system/timers/Timer.cs,307) is pinvoking to win32 apis, replaced with managed equivalent for cross-plat, since these are already implemented in coreclr pal. Measured the accuracy of the pinvoke and the managed apis here, got similar results.